ibis/backends/flink/compiler.py

```diff
@@ -541,3 +541,21 @@ def visit_CountDistinct(self, op, *, arg, where):
         if where is not None:
             arg = self.if_(where, arg, self.f.array(arg)[2])
         return self.f.count(sge.Distinct(expressions=[arg]))
+
+    def visit_MapContains(self, op: ops.MapContains, *, arg, key):
+        return self.f.array_contains(self.f.map_keys(arg), key)
+
+    def visit_Map(self, op: ops.Map, *, keys, values):
+        return self.cast(self.f.map_from_arrays(keys, values), op.dtype)
+
+    def visit_MapMerge(self, op: ops.MapMerge, *, left, right):
+        left_keys = self.f.map_keys(left)
+        left_values = self.f.map_values(left)
+
+        right_keys = self.f.map_keys(right)
+        right_values = self.f.map_values(right)
+
+        keys = self.f.array_concat(left_keys, right_keys)
+        values = self.f.array_concat(left_values, right_values)
+
+        return self.cast(self.f.map_from_arrays(keys, values), op.dtype)
```

Comment:
Just noting here that I can't tell from [the docs for `map_from_arrays`](https://nightlies.apache.org/flink/flink-docs-master/docs/dev/table/functions/systemfunctions/) what the behavior of duplicates keys is here.

Reply:
I also could not find information on the exact behavior of Flink for this. Running the tests, I "inferred" that Flink uses the last value for each key in the order they appear in the array.

Reply:
Makes sense, thanks for looking into it!